sdk/python/kfp/cli/diagnose_me/utility.py

```diff
@@ -0,0 +1,89 @@
+# Lint as: python3
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License,Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Supporting tools and classes for diagnose_me."""
+
+import json
+import subprocess
+from typing import List, Text
+
+
+class ExecutorResponse(object):
+  """Class for keeping track of output of _executor methods.
+
+  Data model for executing commands and capturing their response. This class
+  defines the data model layer for execution results, based on MVC design
+  pattern.
+
+  TODO() This class should be extended to contain data structure to better
+  represent the underlying data instaed of dict for various response types.
+  """
+
+  def execute_command(self, command_list: List[Text]):
+    """Executes the command in command_list.
+
+    sets values for _stdout,_std_err, and returncode accordingly.
+
+    TODO(): This method is kept in ExecutorResponse for simplicity, however this
+    deviates from MVP design pattern. It should be factored out in future.
+
+    Args:
+      command_list: A List of strings that represts the command and parameters
+        to be executed.
+
+    Returns:
+      Instance of utility.ExecutorResponse.
+    """
+
+    try:
+      process = subprocess.Popen(
```

Comment:
AFAIK, [subprocess.run](https://docs.python.org/3/library/subprocess.html#subprocess.run) is a higher-level and easier API to use.